src/ert/config/ert_script.py

```diff
@@ -114,36 +83,70 @@ def initializeAndRun(
         self,
         argument_types: list[type[Any]],
         argument_values: list[str],
-        fixtures: dict[str, Any] | None = None,
+        fixtures: WorkflowFixtures | None = None,
+        **kwargs: dict[str, Any],
     ) -> Any:
         fixtures = {} if fixtures is None else fixtures
-        arguments = []
+        workflow_args = []
         for index, arg_value in enumerate(argument_values):
             arg_type = argument_types[index] if index < len(argument_types) else str
 
             if arg_value is not None:
-                arguments.append(arg_type(arg_value))
+                workflow_args.append(arg_type(arg_value))
             else:
-                arguments.append(None)
-        fixtures["workflow_args"] = arguments
+                workflow_args.append(None)
+
+        fixtures["workflow_args"] = workflow_args
+
+        fixture_args = []
+        all_func_args = inspect.signature(self.run).parameters
+        is_using_wf_args_fixture = "workflow_args" in all_func_args
+
         try:
-            func_args = inspect.signature(self.run).parameters
+            if not is_using_wf_args_fixture:
+                fixture_or_kw_arguments = list(all_func_args)[len(workflow_args) :]
+            else:
+                fixture_or_kw_arguments = list(all_func_args)
+
+            func_args = {k: all_func_args[k] for k in fixture_or_kw_arguments}
+
+            kwargs_defaults = {
+                k: v.default
+                for k, v in func_args.items()
+                if k not in fixtures
+                and v.kind != v.VAR_POSITIONAL
+                and not str(v).startswith("*")
+                and v.default != v.empty
+            }
+            use_kwargs = {
+                k: (kwargs or {}).get(k, default_value)
+                for k, default_value in ({**kwargs_defaults, **kwargs}).items()
```

Comment:
Not sure kwarg support is worth the extra complication, can we do without it?

Reply:
Should be in another PR unless it is needed to fix the bug, think we might need some additional tests for that.

Reply:
Sounds good, yes, can remove the defaulting in the AHMAnalysis job